scripts/pythondistdeps.py

```diff
@@ -11,21 +11,89 @@
 # RPM python dependency generator, using .egg-info/.egg-link/.dist-info data
 #
 
-# Please know:
-# - Notes from an attempted rewrite from pkg_resources to importlib.metadata in
-#   2020 can be found in the message of the commit that added this line.
-
 from __future__ import print_function
 import argparse
-from os.path import basename, dirname, isdir, sep
-from sys import argv, stdin, version
 from distutils.sysconfig import get_python_lib
+from itertools import chain
+from os.path import dirname, sep
+from packaging.requirements import Requirement
+from packaging.version import parse
+import re
+from sys import argv, stdin
 from warnings import warn
 
+try:
+    from importlib.metadata import PathDistribution
+except ImportError:
+    from importlib_metadata import PathDistribution
+
+try:
+    from pathlib import Path
+except ImportError:
+    from pathlib2 import Path
+
+
+class Req(Requirement):
+    def __init__(self, requirement_string):
+        super(Req, self).__init__(requirement_string)
+        self.normalized_name = Req.normalize_name(self.name)
+        self.legacy_normalized_name = Req.legacy_normalize_name(self.name)
+
+    @classmethod
+    def normalize_name(klass, name):
+        """https://www.python.org/dev/peps/pep-0503/#normalized-names"""
+        return re.sub(r'[-_.]+', '-', name).lower()
+
+    @classmethod
+    def legacy_normalize_name(klass, name):
+        """Like pkg_resources Distribution.key property"""
+        return re.sub(r'[-_]+', '-', name).lower()
```

Comment:
Nitpick: I'm not excited about the `klass` name instead of the standard `cls`.

Reply:
I wonder why it is a classmethod and not a staticmethod (when it does not use the "kalss" attribute at all). I would make it a property, but initialized value is fine as well.

Reply:
> I wonder why it is a classmethod and not a staticmethod (when it does not use the "kalss" attribute at all). I would make it a property, but initialized value is fine as well.

It's also used outside of this class, so it can't be a property. But a static method/function makes sense.

Reply:
Ah, good point! staticmethod makes much more sense.